rest/authorization-server/src/main/java/org/eclipse/sw360/rest/authserver/client/rest/OAuthClientController.java

```diff
@@ -0,0 +1,123 @@
+
+    public static final String ENDPOINT_URL = "client-management";
+
+    @Value("${security.oauth2.resource.id}")
+    private String resourceId;
+
+    @Autowired
+    private OAuthClientRepository repo;
+
+    @RequestMapping(method = RequestMethod.GET, path = "")
+    public ResponseEntity<List<OAuthClientResource>> getAllClients() {
+        List<OAuthClientResource> clientResources;
+
+        List<OAuthClientEntity> clients = repo.getAll();
+        if (clients == null) {
+            clients = new ArrayList<>();
+        }
+
+        clientResources = clients.stream().map(OAuthClientResource::new).collect(Collectors.toList());
+
+        return new ResponseEntity<List<OAuthClientResource>>(clientResources, HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.POST, path = "", consumes = "application/json", produces = "application/json")
+    public ResponseEntity<?> createOrUpdateClient(@RequestBody OAuthClientResource clientResource) {
+        OAuthClientEntity clientEntity = null;
+
+        if (StringUtils.isNotEmpty(clientResource.getClientId())) {
+            clientEntity = repo.getByClientId(clientResource.getClientId());
+            if (clientEntity == null) {
+                return new ResponseEntity<String>("No client found for given clientId " + clientResource.getClientId(),
+                        HttpStatus.BAD_REQUEST);
+            }
+        } else {
+            clientEntity = new OAuthClientEntity();
+            clientEntity.setId(UUID.randomUUID().toString().replace("-", ""));
+            clientEntity.setClientId(clientEntity.getId());
+            clientEntity.setClientSecret(UUID.randomUUID().toString());
+        }
+
+        updateClientEntityFromResource(clientEntity, clientResource);
+
+        repo.update(clientEntity);
+
+        return new ResponseEntity<OAuthClientResource>(
+                new OAuthClientResource(repo.getByClientId(clientEntity.getClientId())), HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.DELETE, path = "/{clientId}", consumes = "application/json", produces = "application/json")
+    public ResponseEntity<?> deleteClient(@PathVariable("clientId") String clientId) {
+        OAuthClientEntity clientEntity = null;
+
+        if (StringUtils.isNotEmpty(clientId)) {
+            clientEntity = repo.getByClientId(clientId);
+            if (clientEntity == null) {
+                return new ResponseEntity<String>("No client found for given clientId " + clientId,
+                        HttpStatus.BAD_REQUEST);
+            }
+
+            repo.remove(clientEntity);
+        }
+
+        return new ResponseEntity<OAuthClientResource>(new OAuthClientResource(clientEntity), HttpStatus.OK);
+    }
+
+    private void updateClientEntityFromResource(OAuthClientEntity clientEntity, OAuthClientResource clientResource) {
+        // updateable properties
+        // clientEntity.setClientId(clientResource.getClientId());
```

Comment:
commented code

Reply:
This time on purpose to make clear that those properties have not just been forgotten ^^ But I can remove it and extend the comment.